pkg/operator/ceph/cluster/osd/topology/topology.go

```diff
@@ -148,3 +148,64 @@ func GetDefaultTopologyLabels() string {
 
 	return strings.Join(Labels, ",")
 }
+
+// CheckTopologyConflicts verifies that:
+// 1. No child domain (e.g. rack) has fewer distinct values than its parent (e.g. zone).
+// 2. No topology value is used under more than one label key.
+func CheckTopologyConflicts(nodes []corev1.Node) error {
+	// 1. Build our ordered list of topology labels (region -> zone -> rack -> …), dropping hostname.
+	allLabels := allKubernetesTopologyLabelsOrdered()
+	var topologyHierarchy []string
+	for _, lbl := range allLabels {
+		if lbl != k8sutil.LabelHostname() {
+			topologyHierarchy = append(topologyHierarchy, lbl)
+		}
+	}
+
+	// 2. Gather distinct values for each topology key.
+	topologyValues := make(map[string]map[string]struct{}, len(topologyHierarchy))
+	for _, key := range topologyHierarchy {
+		topologyValues[key] = make(map[string]struct{})
+	}
+	for _, node := range nodes {
+		labels := node.GetLabels()
+		for _, key := range topologyHierarchy {
+			v := client.NormalizeCrushName(labels[key])
+			if v == "" {
+				continue
+			}
+			topologyValues[key][v] = struct{}{}
+		}
+	}
+
+	// 3. Parent-child count check: each child domain must have bigger or equal values than its parent.
```

Comment:
It still doesn't do a full check,

         region
             |
zone1           zone2
   |                     
 rack1

@BlaineEXE wont this be valid hierarchy?
User can have nodes with stale labels?
This algo might not state this a invalid topology

Reply:
cc @travisn ^^